sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -81,6 +86,18 @@ private[spark] class HiveExternalCatalog(client: HiveClient, hadoopConf: Configu
     withClient { getTable(db, table) }
   }
 
+  /**
+   * If the given table properties contains datasource properties, throw an exception.
+   */
```

Comment:
Let's make it clear that this method is used when we want to write metadata to metastore.